### ISSUE TITLE
User is able to create agent tags with more than 20 characters through enroll command.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/elastic-agent#876]
**Kibana version:**  8.4 BC-2 Kibana cloud environment

**Host OS and Browser version:** All, All

**Build details:**
```
VERSION: 8.4.0
BUILD: 55166
COMMIT: 9e9e0d6a685cbc2858a85a357f93dcb76259fdee
```

**Preconditions:**
1. 8.4 BC2 Kibana cloud environment should be available.

**Steps to reproduce:**
1. Login to Kibana environment.
2. Install an agent with say --tag= `abcabcabcabcabcabcabcabcabcabc1234567890` (More than 20 characters.)
3. Observe user is able to create tag with more than 20 characters.

**Expected Result:**
As per PR https://github.com/elastic/kibana/pull/136019#issue-1299131578 it is expected that tag name should be limited to 20 characters.

**Screenshots:**
![1](https://user-images.githubusercontent.com/77374876/183354160-cd7cd0f0-f9b1-49b3-8ee0-4534bd7c9796.jpg)
![2](https://user-images.githubusercontent.com/77374876/183354166-68434341-c26c-4e30-8f05-6d18b1a45897.JPG)

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Secondary review for this ticket is Done

**Comment by amolnater-qasource** [comment]
Hi Team,

We have revalidated this issue on latest 8.10.0 SNAPSHOT kibana cloud environment and found it **still reproducible**.

**Observations:**
- User is able to create agent tags with more than 20 characters tags with install command.


**Screenshot:**
![image](https://github.com/elastic/elastic-agent/assets/77374876/b151e75e-3487-4dd4-9d8f-04298376acd2)
![image](https://github.com/elastic/elastic-agent/assets/77374876/2e095032-0c44-4054-994b-fb641751dd86)



**Build details:**
VERSION: 8.10.0 SNAPSHOT
BUILD: 65561
COMMIT: 4dadcbb91136a3e7d39e4b419aa12e21a37e4918


Please let us know if anything else is required from our end.

Thanks!

**Comment by pierrehilbert** [comment]
As this sanitize is done at Kibana level, should be assigned to Fleet Team @jen-huang / @kpollich

**Comment by jlind23** [comment]
@pierrehilbert tag are supplied at the enrolment. Not sure Elastic Agent prevents you from using more than 20chars.. The only option for Kibana would be to truncate these tags then..

**Comment by pierrehilbert** [comment]
Yes, that's the part of the code @juliaElastic wrote and @amolnater-qasource mentioned.
Will probably be better to have this check during enrollment but still a problem with the Sanitize function currently in place.

**Comment by jen-huang** [comment]
I'm moving this to fleet server. The sanitization code referenced is on Kibana send when tags are added/modified through UI, not adding during enrollment. We should have same set of sanitization on fleet server.
